card/mod/01_history/set/all/actions.rb

```diff
@@ -0,0 +1,47 @@
+# -*- encoding : utf-8 -*-
```

Comment:
They are not the same, but connected.  The full dependency feature would have to handle lots of stuff eventually, including mods from different containers.  You are talking the analogy to a gemspec for mods, which supports Gemfile functionality.  Seems to me that we would do selection and ordering in the same features, and if you have one, you'll want the other pretty soon.

The real issue here is the set of modules included in the core gem are just all loaded, so there is no dependency handling involved, just include all of these is this order.  Once we re-organize these standard and core mods into a handful of dependency chains, we'll want to be able to 1) not include some chains, or 2) replace a dependency with a mod handling the same role.

The history module added in this change is meant to be replaceable in this way.  I might have a gem containing one or more mods that implements externalized history.  One replaces the standard history module with one that sends messages to listeners, and several listeners that can process the messages to store them, send them to networked clients or perform notifications and other post processing.